test/integration/src/test/java/io/pravega/test/integration/SegmentReaderTest.java

```diff
@@ -0,0 +1,237 @@
+    public void testSegmentReadOnSealedStream() throws ExecutionException, InterruptedException {
+        String scope = "testSegmentReaderScope";
+        String stream = "testSegmentReaderStream";
+        int noOfEvents = 50;
+        long timeout = 1000;
+        int readEventCount = 0;
+        createStream(scope, stream, 1);
+        writeEventsIntoStream(noOfEvents, scope, stream);
+
+        log.info("Creating segment reader manager.");
+        @Cleanup
+        SegmentReaderManager<Integer> segmentReaderManager = SegmentReaderManager.create(clientConfig, serializer);
+        List<SegmentReader<Integer>> segmentReaderList = segmentReaderManager.getSegmentReaders(Stream.of(scope, stream), null).get();
+        assertEquals(1, segmentReaderList.size());
+
+        boolean isSealed = controllerWrapper.getController().sealStream(scope, stream).join();
+        assertTrue("isSealed", isSealed);
+
+        SegmentReader<Integer> segmentReader = segmentReaderList.get(0);
+        log.info("Starting reading the events.");
+        while (true) {
+            try {
+                segmentReader.read(timeout);
+                readEventCount++;
+            } catch (EndOfSegmentException e) {
+                break;
+            }
+        }
+        log.info("Reading of events is successful.");
+        assertEquals(noOfEvents, readEventCount);
+    }
+
+    @Test(timeout = 50000)
+    public void testSegmentReadWithTruncatedStream() throws ExecutionException, InterruptedException {
+        String scope = "testSegmentReaderWithTruncatedScope";
+        String stream = "testSegmentReaderWithTruncatedStream";
+        int noOfEvents = 10;
+        long timeout = 1000;
+        int readEventCount = 0;
+        createStream(scope, stream, 1);
+        writeEventsIntoStream(noOfEvents, scope, stream);
+
+        StreamCut streamCut = new StreamCutImpl(Stream.of(scope, stream), ImmutableMap.of(new Segment(scope, stream, 0L), 60L));
+        log.info("Truncating two events from stream");
+        boolean isTruncated = controllerWrapper.getController().truncateStream(scope, stream, streamCut).join();
+        assertTrue("isTruncated", isTruncated);
+
+        log.info("Sealing stream");
+        boolean isSealed = controllerWrapper.getController().sealStream(scope, stream).join();
+        assertTrue("isSealed", isSealed);
+
+        streamCut = new StreamCutImpl(Stream.of(scope, stream), ImmutableMap.of(new Segment(scope, stream, 0L), 0L));
+
+        log.info("Creating segment reader manager.");
+        @Cleanup
+        SegmentReaderManager<Integer> segmentReaderManager = SegmentReaderManager.create(clientConfig, serializer);
+        //Try to read the segment from offset 0.
+        List<SegmentReader<Integer>> segmentReaderList = segmentReaderManager.getSegmentReaders(Stream.of(scope, stream), streamCut).get();
+        assertEquals(1, segmentReaderList.size());
+        SegmentReader<Integer> segmentReader = segmentReaderList.get(0);
```

Comment:
Why Integer ?

Reply:
Thanks, i have updated the Serializer and also add assert to verify the events.